### PR TITLE
Add MIPI STPv2 decoder

### PR DESCRIPTION
**Pull request by nordic-krch**
Add a module which decodes data encoded into MIPI STPv2.
Test is also added.
Module is placed in `subsys/debug/coresight` because Coresight STM is using it but it is actually not ARM specific so it is to decide where to put it finally (`subsys/debug/mipi`?)
It relates to #72129 which provides Coresight Trace deformatter and #72091 which provides dictionary-based logging on nrf54h20 using Coresight STM. In dictionary-based logging data is decoded by the host application but it will be also possible to decode STM data on chip using this decoder (this PR), coresight trace deformatter (#72129) and #72476 which has full solution.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

File: subsys/debug/mipi_stp_decoder.c [hunk 1]
```diff
@@ -0,0 +1,791 @@
...
+}
+
+/* Function swaps nibbles in a byte. */
+static uint8_t swap8(uint8_t byte)
+{
+	return (byte << 4) | (byte >> 4);
+}
+
+/* Function swaps nibbles in a 16 bit variable. */
+static uint16_t swap16(uint16_t halfword)
```
Comment (kl-cruz) [inline subsys/debug/mipi_stp_decoder.c:563]: I'd recommend to change names of these functions to for example swap_nibbles8/16/32 to reflect functionality in direct way

**Review by dcpleung — DISMISSED** [review]

**Comment by nashif** [comment]
@nordic-krch  can you please add some documentation?

**Review by dcpleung — APPROVED** [review]

**Comment by nordic-krch** [comment]
@nashif Documentation added.

**Review by nashif — APPROVED** [review]

File: include/zephyr/debug/mipi_stp_decoder.h [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup mipi_stp_decoder_apis STP Decoder API
+ * @ingroup coresight_apis
```
Comment (kartben) [inline include/zephyr/debug/mipi_stp_decoder.h:18]: @nordic-krch we don't have such a doxygen group in Zephyr upstream -- was this an oversight? Can you please fix this by either adding `mipi_stp_decoder_apis` to an existing Zephyr group, or by defining a new one?

**Comment by nordic-krch** [comment]
@kartben I aligned decoder to 2.4 in #72476.

**Pull request merged**